### ISSUE TITLE
Custom form XML schema definitions ignored at validation

### ISSUE THREAD

**Issue by R3dLantern** [sulu/sulu#6439]
| Q | A
| --- | ---
| Sulu Version | 2.3.9
| PHP Version | 8.0.3
| DB Version | 10.4.12-MariaDB
| Browser Version | Chrome 96.0.4664.110 64-bit

# Actual Behavior

When defining an `anyOf` schema condition in a form XML of a custom entity, it is being ignored in the form validation.

# Expected Behavior

An analysis of `Sulu\PageBundle\Resources\config\forms\page_settings.xml` shows, among others, the following configuration:
```xml
<schema>
  <allOf>
    <!-- ... -->
    <schema>
      <anyOf>
        <schema>
          <properties>
            <property name="shadowOn" value="true" />
            <property name="shadowBaseLanguage" mandatory="true" />
          </properties>
          </schema>
          <schema>
          <properties>
            <property name="shadowOn" value="false" />
          </properties>
        </schema>
      </anyOf>
    </schema>
    <!-- ... -->
  </allOf>
</schema>
```
With `shadowOn` being a `checkbox` and `shadowBaseLanguage` not being a mandatory value as of its property configuration, I deduced that the schema configuration above represents the validation condition that `shadowBaseLanguage` becomes a mandatory field as soon as `shadowOn` is checked.
As such, I suppose these `<schema>` definitions should be considered in the form validation on each `FinishFieldHandler`s.
A test with the rendered `analytic_details` form representation supports my conclusion, as the form triggers an error highlight when e.g. omitting a Google Tag Manager key.

# Steps to Reproduce

Define a custom entity form. Notably, all properties that are evaluated in the schema condition are nested in a section.
```xml
<?xml version="1.0" ?>
<form xmlns="http://schemas.sulu.io/template/template"
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
    xsi:schemaLocation="http://schemas.sulu.io/template/template http://schemas.sulu.io/template/form-1.0.xsd"
>
  <key>foo_details</key>
  <schema>
    <allOf>
      <schema>
        <anyOf>
          <schema>
            <properties>
              <property name="fizz" value="true"/>
              <property name="buzz" mandatory="true"/>
            </properties>
          </schema>
          <schema>
            <properties>
              <property name="fizz" value="false"/>
            </properties>
          </schema>
        </anyOf>
      </schema>
    </allOf>
  </schema>
  <properties>
    <section name="bar">
      <meta>
        <title>app.foo.bar</title>
      </meta>
      <properties>
        <property name="fizz" type="checkbox">
          <meta>
            <title>app.foo.fizz</title>
          </meta>
        </property>
        <property name="buzz" type="text_line" visibleCondition="fizz == true">
          <meta>
            <title>app.foo.buzz</title>
          </meta>
        </property>
      </properties>
    </section>
  </properties>
</form>
```
Another test without section nesting showed the same effect.

**Comment by niklasnatter** [comment]
When inspecting the request to `/admin/metadata/form/foo_details`, you can see that the form configuration leads to the following json schema:
```
"anyOf": [
    {
        "type": "object",
        "properties": {
            "fizz": {
                "const": true
            }
        },
        "required": [
            "buzz"
        ]
    },
    {
        "type": [
            "number",
            "string",
            "boolean",
            "object",
            "array",
            "null"
        ]
    }
]
```

I think this is caused because the following check in the `SchemaXmlParser` does not correctly handle `false` as value:
https://github.com/sulu/sulu/blob/c3d009cdd1809590ea12a6de497f97ebf2d7cf89/src/Sulu/Component/Content/Metadata/Parser/SchemaXmlParser.php#L80

**Comment by niklasnatter** [comment]
Should be fixed in #6459, thanks for reporting 🙂
